### PR TITLE
improve audit score

### PR DESCRIPTION
**Pull request by kmdrGroch**

### CODE REVIEW

**Comment by nicojs** [comment]
First of, thanks for joining our efforts on the redesign branch! I wasn't looking forward to the upcoming merge conflict when merging your other PR to master :).

Could you explain to me how the caching of the service worker works? When will it invalidate the cache? Would we need to version our JavaScript with a hash of the fil content?

**Comment by kmdrGroch** [comment]
@nicojs service worker caches data in your fs. You can check if it created cache by going (in chrome), F12 -> application -> cache storage. As you can see in sw.js there is an array with files to be cached like also a version. After changing a version, cache will be cleared and files will be cached again (or after clearing cache by yourself, so ctrl + shift + r should remove cache). My advice for it -> after every release commit, just bump up a version in sw.js and add / remove / stay files in array and everything should be good :)

File: sw.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        })
+      );
+    })
+  );
+});
+
+// Call Fetch Event
+self.addEventListener('fetch', e => {
+  console.log('Service Worker: Fetching');
+  e.respondWith(fetch(e.request).catch(() => caches.match(e.request)));
```
Comment (hugo-vrijswijk) [inline sw.js]: Doesn't this first try to fetch, and then attempt to get from cache if the request fails. Wouldn't we want it the other way around, so the cache is called first?

Comment (kmdrGroch) [reply sw.js]: well @hugo-vrijswijk, you may have a point. Now it works like -> check if there is connection -> if yes, load using web, if not try get cache.

**Comment by kmdrGroch** [comment]
done @hugo-vrijswijk

**Comment by hugo-vrijswijk** [comment]
What is the best way to go about updating the cache version? Every release is probably a good idea. But also something that is probably very easy to forget. Is there a way to do this automatically? Maybe use the commit hash and somehow include it in the build?

**Comment by kmdrGroch** [comment]
@hugo-vrijswijk I can think about it, but I don't see any nice posible way. We could just create a file after each commit and replace it, with a newer version (do some template, lets say). It is not hard to do and if you think its a good idea i can make it so.

**Comment by kmdrGroch** [comment]
@nicojs, @hugo-vrijswijk after running ``npm run build`` it updates service worker automatically. (bumps version in package.json too [only last digit])

**Comment by nicojs** [comment]
Awesome! I'll take a look tomorrow morning. Great job

**Comment by kmdrGroch** [comment]
as you probably noticed / or not. There is also new PR in stryker (small 1 / 1 docs update), and refactor of ``stryker-cli`` (now it needs node >=6, but all dependencies are up to date [in audit there was like 60 issues, 1 critical, ~5 high]) [https://docs.npmjs.com/getting-started/about-audit-reports]

File: tasks/createCache.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+packageJSON.version = version;
+packageJSON = JSON.stringify(packageJSON, null, 2);
+
+/**
+ * @param { string } dir
+ * @param { Array } filelist
+ * @returns { Array } all html files, all.css and all.bundle.js
+ */
+const walkSync = (dir, filelist = []) => {
```
Comment (nicojs) [inline tasks/createCache.js]: Could we replace this with [glob](https://npmjs.org/package/glob)? That would be much more readable.

Comment (kmdrGroch) [reply tasks/createCache.js]: It bumps package.json version coz cache version is taken from there.

**Comment by nicojs** [comment]
Thanks for the PR. Let's work on the service worker together in a branch. If you make a start I'll make sure it works nicely with Grunt 👍

**Comment by kmdrGroch** [comment]
Should I make separate branch for it or PR it to redesign branch?

**Comment by nicojs** [comment]
Redesign branch is fine. I'm planning to create a build with publication in that branch. Let's keep it there.

**Pull request merged**